### PR TITLE
Add json schema validation for transaction.signatures property - Closes #1845

### LINKED ISSUE #1845: Post mutisig tx via peers with invalid signatures takes down the node
### Expected behavior

Node rejects the transaction.

### Actual behavior
Application fails here:
https://github.com/LiskHQ/lisk/blob/7bf54a05815049f31931ffaba379b6d3b5febe79/logic/transaction.js#L671

```
[dbg] 2018-03-26 13:31:32 | Received transaction 15193251105819760810 from peer 127.0.0.1:4000
[FTL] 2018-03-26 13:31:32 | System error - {"message":"Invalid hex string","stack":"TypeError: Invalid hex string\n    at TypeError (native)\n    at Buffer.write (buffer.js:760:21)\n    at fromString (buffer.js:203:26)\n    at Function.Buffer.from (buffer.js:99:12)\n    at Transaction.verifyBytes (/Users/diego-g/git/lisk-0.9/logic/transaction.js:683:32)\n    at Transaction.verifySignature (/Users/diego-g/git/lisk-0.9/logic/transaction.js:624:14)\n    at Transaction.verify (/Users/diego-g/git/lisk-0.9/logic/transaction.js:557:14)\n    at verifyTransaction (/Users/diego-g/git/lisk-0.9/logic/transactionPool.js:700:30)\n    at nextTask (/Users/diego-g/git/lisk-0.9/node_modules/async/dist/async.js:5070:14)\n    at Immediate.<anonymous> (/Users/diego-g/git/lisk-0.9/node_modules/async/dist/async.js:5064:13)"}
```

### Steps to reproduce
Posting a transaction through websockets or HTTP layer as well.

### Which version(s) does this affect? (Environment, OS, etc...)
`0.9.12` & `1.0.0`. Already solved in `0.9.13`

### PR DESCRIPTION
**Pull request by diego-G**
### What was the problem?
We have no json schema for `transactions.signatures` array. This can be used by an user who wants to sign offline a multisig transaction and broadcast it with some or all the signatures.

### How did I fix it?
Adding json schema validation for `transaction.signatures` property.

### How to test it?
`test/functional/http/post/4.multisignature.advanced.js`

### Review checklist

* The PR solves #1845 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: schema/swagger.yml [hunk 1]
```diff
@@ -1175,7 +1175,7 @@ definitions:
         type: string
         format: signature
         example: 2821d93a742c4edf5fd960efad41a4def7bf0fd0f7c09869aed524f6f52bf9c97a617095e2c712bd28b4279078a29509b339ac55187854006591aa759784c205
-      multisignatures:
```
Comment (nazarhussain) [inline schema/swagger.yml]: @diego-G If this change is aligned with LiskJS then its good to go.

Comment (diego-G) [reply schema/swagger.yml]: @willclarktech @shuse2 please if you aren't yet, be aware of it

Comment (willclarktech) [reply schema/swagger.yml]: @diego-G Can you summarise the change we need to implement?

Comment (diego-G) [reply schema/swagger.yml]: As we discussed, there you go @willclarktech https://github.com/LiskHQ/lisk-js/issues/640

**Pull request merged**